LiskHQ/lisk-sdk
# Create http api plugin setup - Closes #5248

### What was the problem?

This PR resolves #5248 

### How was it solved?

- Add new folder structure for the plugins
- Rename BaseModule to BasePlugin
- Adjust framework interface to be able to compile
- Add simple unit test for the middleware
- Add sample endpoint
- Add empty swagger yaml

### How was it tested?
- Link with core with branch https://github.com/LiskHQ/lisk-sdk/pull/5458
- Start the devnet
- Check if the endpoint is available and returning properly
